lib/rails_core_extensions/translations.rb

```diff
@@ -0,0 +1,44 @@
+module RailsCoreExtensions
+  module Translations
+    def self.included(base)
+      base.extend ClassMethods
+    end
+
+    module ClassMethods
+      def translate(key, options = {})
+        I18n.translate key, options.merge(scope: translation_key)
+      end
+
+      def translation_key
+        (base_activerecord_class || base_ruby_class).name.tableize.singularize
+      end
+
+      def base_activerecord_class
+        return unless defined?(ActiveRecord) && is_a?(ActiveRecord::Base)
+        klass = self
+        while klass.superclass != ActiveRecord::Base
+          klass = klass.superclass
+        end
+        klass
+      end
+
+      def base_ruby_class
```

Comment:
I want to just climb the superclasses for efficiency reasons, e.g. most of the time you will just call superclass once, on STI it might be twice, whereas ancestors is a bit more indepth.